### PR TITLE
Add support for printing reagents in lathes

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

for future PRs.

also cleans up some old old lathe shit that sucked.

## Technical details

adds new ResultReagents to latherecipe so that you can now have recipes that spit out custom stuff. also restores the icon field which broke with some recent changes.

## Media

https://github.com/user-attachments/assets/f98c0e96-0091-471f-99a8-3a1dd62b9874

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.Shared/Research/Prototypes/LatheRecipePrototype.cs [hunk 1]
```diff
@@ -1,101 +1,58 @@
...
-
-
-        /// <summary>
-        ///     How many milliseconds it'll take for the lathe to finish this recipe.
-        ///     Might lower depending on the lathe's upgrade level.
-        /// </summary>
-        [ViewVariables]
-        public TimeSpan CompleteTime => _completeTime;
+        [DataField]
+        public Dictionary<ProtoId<MaterialPrototype>, int> Materials = new();
```
Comment (0x6273) [inline Content.Shared/Research/Prototypes/LatheRecipePrototype.cs:53]: Why was this renamed?

Comment (EmoGarbage404) [reply Content.Shared/Research/Prototypes/LatheRecipePrototype.cs:53]: the general convention nowadays is to have variables match the datafield string exactly. its a relatively minor change and doesn't affect any yaml so i thought it'd just be good to update while i was working with everything else.
